docs/csharp/whats-new/csharp-8.md

```diff
@@ -0,0 +1,314 @@
+
+```csharp
+public static decimal ComputeSalesTax(Address location, decimal salePrice) =>
+    location switch
+{
+    { State: "WA" } => salePrice * 0.06M,
+    { State: "MN" } => salePrice * 0.75M,
+    { State: "MI" } => salePrice * 0.05M,
+    // other cases removed for brevity...
+    _ => 0M
+};
+```
+
+Pattern matching creates a concise syntax for expressing this algorithm.
+
+### Tuple patterns
+
+Some algorithms depend on multiple inputs. **Tuple patterns** allow you to switch based on multiple values expressed as a [tuple](../tuples.md).  The following code shows a switch expression for the game *rock, paper, scissors*:
+
+```csharp
+public static string RockPaperScissors(string first, string second)
+    => (first, second) switch
+{
+    ("rock", "paper") => "rock is covered by paper. Paper wins.",
+    ("rock", "scissors") => "rock breaks scissors. Rock wins.",
+    ("paper", "rock") => "paper covers rock. Paper wins.",
+    ("paper", "scissors") => "paper is cut by scissors. Scissors wins.",
+    ("scissors", "rock") => "scissors is broken by rock. Rock wins.",
+    ("scissors", "paper") => "scissors cuts paper. Scissors wins.",
+    (_, _) => "tie"
+};
+```
+
+The messages indicate the winner. The discard case represents the three combinations for ties, or other text inputs.
+
+### Positional patterns
+
+Some types include a `Deconstruct` method that deconstructs its properties into discrete variables. When a `Deconstruct` method is accessible, you can use **positional patterns** to inspect properties of the object and use those properties for a pattern.  Consider the following `Point` class that includes a `Deconstruct` method to create discrete variables for `X` and `Y`:
+
+```csharp
+public class Point
+{
+    public int X { get; }
+    public int Y { get; }
+
+    public Point(int x, int y) => (X, Y) = (x, y);
+
+    public void Deconstruct(out int x, out int y) =>
+        (x, y) = (X, Y);
+}
+```
+
+The following method uses the **positional pattern** to extract the values of `x` and `y`. Then, it uses a `when` clause to determine the quadrant of the point:
+
+```csharp
+static string Quadrant(Point p) => p switch
+{
+    (0, 0) => "origin",
+    (var x, var y) when x > 0 && y > 0 => "Quadrant 1",
+    (var x, var y) when x < 0 && y > 0 => "Quadrant 2",
+    (var x, var y) when x < 0 && y < 0 => "Quadrant 3",
+    (var x, var y) when x > 0 && y < 0 => "Quadrant 4",
+    (var x, var y) => "on a border",
+    _ => "unknown"
+};
+```
+
+## using declarations
+
+A **using declaration** is a variable declaration preceded by the `using` keyword. It tells the compiler that the variable being declared should be disposed at the end of the enclosing scope. For example, consider the following code that writes a text file:
+
+```csharp
```

Comment:
In the previous pattern matching section you used the format:
```
NEW

compare to

OLD
```

but in this one you are starting with the old way of doing it and showing the new way. I would rather see the new way upfront because most likely everyone is familiar with how to write a `using` statement and would rather just see the change first.